cmd/watch-termination/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+func main() {
+	terminationLog := flag.String("termination-log-file", "", "Write logs after SIGTERM to this file (in addition to stderr)")
+	terminationLock := flag.String("termination-touch-file", "", "Touch this file on SIGTERM and delete on termination")
+
+	flag.Parse()
+	args := flag.CommandLine.Args()
+
+	if len(args) == 0 {
+		fmt.Println("Missing command line")
+		os.Exit(1)
+	}
+
+	termCh := make(chan struct{})
+	var stderr io.Writer = os.Stderr
+	if len(*terminationLog) > 0 {
+		stderr = &eventuallyTeeWriter{
+			Writer:             os.Stderr,
+			fn:                 *terminationLog,
+			startFileLoggingCh: termCh,
+		}
+	}
+
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = stderr
+
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGTERM, syscall.SIGINT)
```

Comment:
are there more termination signals to handle? e.g. SIGABRT?

Reply:
SIGABRT is handled by Golang. You destroy debugging abilities if you trap it.